File: frontend/lib/models/zingsnis_template.dart
class ZingsnisTemplate {
  final int id;
  String pavadinimas;
  String aprasymas;
  int testasId;
  int eile;
  bool komentarasPrivalomas;
  bool nuotraukaPrivaloma;

  ZingsnisTemplate({
    required this.id,
    required this.pavadinimas,
    required this.aprasymas,
    required this.testasId,
    required this.eile,
    this.komentarasPrivalomas = false,
    this.nuotraukaPrivaloma = false,
  });

  factory ZingsnisTemplate.fromJson(Map<String, dynamic> j) => ZingsnisTemplate(
        id: j['id'] as int,
        pavadinimas: j['pavadinimas'] ?? '',
        aprasymas: j['aprasymas'] ?? '',
        testasId: j['testasId'] ?? j['testas_id'] ?? 0,
        eile: j['eile'] ?? 0,
      komentarasPrivalomas: (j['komentarasPrivalomas'] ?? j['komentaras_privalomas'] ?? false) as bool,
      nuotraukaPrivaloma: (j['nuotraukaPrivaloma'] ?? j['nuotrauka_privaloma'] ?? false) as bool,
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'pavadinimas': pavadinimas,
        'aprasymas': aprasymas,
        'testasId': testasId,
        'eile': eile,
      'komentarasPrivalomas': komentarasPrivalomas,
      'nuotraukaPrivaloma': nuotraukaPrivaloma,
      };
}

File: Backend/Models/ZingsnisTemplate.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Models
{
    [Table("zingsnis_template")]
    public class ZingsnisTemplate
    {
        [Key]
        [Column("id")]
        public int Id { get; set; }

        [Required]
        [Column("pavadinimas")]
        public string Pavadinimas { get; set; } = null!;

        [Required]
        [Column("aprasymas")]
        public string Aprasymas { get; set; } = null!;

        [Required]
        [Column("testas_id")]
        public int TestasId { get; set; }

        [Required]
        [Column("eile")]
        public int Eile { get; set; }
        
        [Column("komentaras_privalomas")]
        public bool KomentarasPrivalomas { get; set; } = false;

        [Column("nuotrauka_privaloma")]
        public bool NuotraukaPrivaloma { get; set; } = false;

        [JsonIgnore]
        public Testas? Testas { get; set; }
    }
}
